.github/workflows/python_tests.yml

```diff
@@ -43,6 +43,9 @@ jobs:
         run: |
           python -m pip install --upgrade pip
           pip install .[dev]
+          
+      - name: Lint with ruff
```

Comment:
Any reason this needs to be done as part of the python test workflow, what about doing it on its own like `darker`?

Also, do we still need `darker` if `ruff` is now being used?

Reply:
> Any reason this needs to be done as part of the python test workflow, what about doing it on its own like darker?

I added into here because this workflow already installs a heap of stuff and it seemed more efficient to add one rather than creating a whole new image build. Happy to split it out if that's what's preferred.

> Also, do we still need darker if ruff is now being used?

Yes, they do very different things. darker is a formatter, ruff is a linter. The former only cares about code layout; the latter cares about the code itself and looks for common anti-patterns.

Reply:
> I added into here because this workflow already installs a heap of stuff and it seemed more efficient to add one rather than creating a whole new image build. Happy to split it out if that's what's preferred.

The path I'm going with this question is if we don't do it as part of python tests, is it even needed in the dependency requirements. Is this the only reason you need to add `ruff` to the dependency requirements? It's like `darker` - we don't need a dependency on `darker`, it's only used in pre-commit and CI, and not in the code itself.

Reply:
👍 I think if we split it out, then we don't need to update the Pipfile and requirements files, or `pyproject.toml` which is a lot simpler.

Taking a step back, `ruff` isn't really a `nucypher` dependency, but rather a tool used for builds/pre-commit hooks, so splitting it out seems cleaner to me.

Reply:
OK. I'll get that done at some point...